### DISCUSSION TITLE
Subscribe API difference in v11

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10377] [General]
[MediaPlayer.Avalonia](https://github.com/mysteryx93/MediaPlayerUI.NET) breaks in Avalonia v11. 

    newValue.GetObservable(PlayerHostBase.PositionProperty).Subscribe(Player_PositionChanged);

now says 

    Argument type 'method group' is not assignable to parameter type 'System.IObserver<System.TimeSpan>'

What's the correct way of solving this?

Also is there documentation of how to upgrade old styles to new styles for 3rd party controls?

**Comment by jp2masa** [comment]
Similar to https://github.com/AvaloniaUI/Avalonia/discussions/10223#discussioncomment-4873946.

In this case, although the `Subscribe` extension method you are using is in the `System` namespace, it is actually defined in the `System.Reactive` assembly.

**Comment by mysteryx93** [comment]
OK adding `System.Reactive` is a pretty simple solution!

Project runs.

Strangely, MediaPlayerUI is broken in Avalonia11. The seek slider work perfectly in terms of behaviors... but in appearance, the thumb always stays at 0 ??? Bug in Avalonia?

**Reply by maxkatz6** [reply]
Probably bug in custom styles. It works fine in current master and IIRC in preview5 too.

**Reply by mysteryx93** [reply]
Styles are written using the old ways; where is information about how to upgrade the styles to the new framework?

**Reply by maxkatz6** [reply]
Not yet. If previously you have copied 0.10 styles and adapted in your code, then you can do the same with 11.0 styles.
Or, most likely the specific reason is mismatching x:Name values. As it was more standardized in 11.0. Note that all "PART_" named control must be presented in your templates to keep control working properly.

**Reply by mysteryx93** [reply]
There was also some changes about 3rd party styles being more modular or something? Ah ya a name change would make sense why it works but visuals don't follow.

**Reply by maxkatz6** [reply]
> There was also some changes about 3rd party styles being more modular or something

Not much. If you use ControlTheme for your controls, then it is easier to change or replace for everybody else. As before you could only stack styles on top of previous ones (or remove whole style includes). That's true.

**Reply by mysteryx93** [reply]
`thumb` was renamed to `SliderThumb`. Doesn't fix it though. Also in the Fluent template, it is still `thumb` in `DataValidationErrors`, looks like an error.

Would have to look deeper for my issue. Also my volume control is completely funky.

I need `PART_DecreaseButton`, `PART_IncreaseButton` and `SliderThumb`; anything else is mandatory? [My style.](https://github.com/mysteryx93/MediaPlayerUI.NET/blob/master/Avalonia/Styles/Default/Theme.axaml)

**Reply by mysteryx93** [reply]
OK I reviewed with DevTools. There really hasn't been much difference other than adopting ControlThemes. The problem is that my Track control does not have Minimum, Maximum and Value properties set so although they are set on the Slider, they remain default on the Track. The name `PART_Track` is correct though.

Does it absolutely require `PART_Track` to be part of a ControlTheme to be found? I'm not sure if ControlTheme API is stable, there's no doc for it, and I'm not sure on the exact way of converting my code for it the right way.

**Reply by timunie** [reply]
Yes it requires all PART named controls to be part of the ControlTemplate. Wether this is inside a Style or ControlTheme shouldn't matter.

**Reply by mysteryx93** [reply]
The problem was that this was added to the style; copy/pasted it and now it works. Explains why Minimmum, Maximum and Value were not passed over!

```xaml
   <Style Selector="^ /template/ Track#PART_Track">
      <Setter Property="Minimum" Value="{TemplateBinding Minimum}" />
      <Setter Property="Maximum" Value="{TemplateBinding Maximum}" />
      <Setter Property="Value" Value="{TemplateBinding Value, Mode=TwoWay}" />
    </Style>
    <Style Selector="^ /template/ Border#TrackBackground">
      <Setter Property="BorderThickness" Value="2" />
      <Setter Property="BorderBrush" Value="{DynamicResource ThemeBorderLowBrush}" />
    </Style>
    <Style Selector="^ /template/ TickBar">
      <Setter Property="Ticks" Value="{TemplateBinding Ticks}" />
    </Style>
    <Style Selector="^:disabled /template/ Grid#grid">
      <Setter Property="Opacity" Value="{DynamicResource ThemeDisabledOpacity}" />
    </Style>
```
